src/main/kotlin/io/github/gradlenexus/publishplugin/InitializeNexusStagingRepository.kt

```diff
@@ -48,7 +49,10 @@ open class InitializeNexusStagingRepository @Inject constructor(
         val client = NexusClient(serverUrl, repository.username.orNull, repository.password.orNull, clientTimeout.orNull, connectTimeout.orNull)
         val stagingProfileId = determineStagingProfileId(repository, client)
         logger.info("Creating staging repository for {} at {}, stagingProfileId '{}'", repository.name, serverUrl, stagingProfileId)
-        registry.get()[repository.name] = client.createStagingRepository(stagingProfileId)
+        val descriptor = client.createStagingRepository(stagingProfileId)
+        val consumerUrl = HttpUrl.get(serverUrl)!!.newBuilder().addEncodedPathSegments("repositories/${descriptor.stagingRepositoryId}/content/").build()
+        logger.lifecycle("Created staging repository '{}' at {}", descriptor.stagingRepositoryId, consumerUrl)
```

Comment:
I wonder about the logging level. With `lifecycle` it is printed in every build and it "breaks" the default "non-verbosity" promoted in the Gradle (that a task is not printed by default if it doesn't output anything). On the other hang, `info` is too verbose (and cannot be enabled per tasks) and we rather would not want to add a separate switch to turn it on and off. Therefor, I would probably leave it as is.

Reply:
In this case I wouldn't consider the output "noise" since the URL is useful for testing the deployed artifact.

Reply:
Useful, for the projects which do that testing :-)